src/viadot/sources/mediatool.py

```diff
@@ -0,0 +1,334 @@
+            media_type_ids (list[str]): List of media type IDs.
+
+        Returns:
+            list[dict[str, str]]: A list of dicts will be returned.
+        """
+        list_media_types = []
+        for id_media_type in media_type_ids:
+            response = handle_api_response(
+                url=f"https://api.mediatool.com/mediatypes/{id_media_type}",
+                headers=self.header,
+                method="GET",
+            )
+            response_dict = json.loads(response.text)
+            list_media_types.append(
+                {
+                    "_id": response_dict.get("mediaType").get("_id"),
+                    "name": response_dict.get("mediaType").get("name"),
+                    "type": response_dict.get("mediaType").get("type"),
+                }
+            )
+
+        return list_media_types
+
+    def api_connection(
+        self,
+        get_data_from: str,
+        organization_id: str | None = None,
+        vehicle_ids: list[str] | None = None,
+        media_type_ids: list[str] | None = None,
+    ) -> list[dict[str, str]]:
+        """General method to connect to Mediatool API and generate the response.
+
+        Args:
+            get_data_from (str): Method to be used to extract data from.
+            organization_id (str, optional): Organization ID. Defaults to None.
+            vehicle_ids (list[str]): List of organization IDs. Defaults to None.
+            media_type_ids (list[str]): List of media type IDs. Defaults to None.
+
+        Returns:
+            list[dict[str, str]]: Data from Mediatool API connection.
+        """
+        self.url_abbreviation = get_data_from
+
+        if self.url_abbreviation == "organizations":
+            returned_data = self._get_organizations(self.user_id)
+
+        elif self.url_abbreviation == "media_entries":
+            returned_data = self._get_media_entries(organization_id=organization_id)
+
+        elif self.url_abbreviation == "vehicles":
+            returned_data = self._get_vehicles(vehicle_ids=vehicle_ids)
+
+        elif self.url_abbreviation == "campaigns":
+            returned_data = self._get_campaigns(organization_id=organization_id)
+
+        elif self.url_abbreviation == "media_types":
+            returned_data = self._get_media_types(media_type_ids=media_type_ids)
+
+        return returned_data
+
+    @add_viadot_metadata_columns
+    def to_df(
+        self,
+        if_empty: str = "warn",
+        **kwargs,
+    ) -> pd.DataFrame:
+        """Pandas Data Frame with the data in the Response object and metadata.
+
+        Args:
+            if_empty (str, optional): What to do if a fetch produce no data.
+                Defaults to "warn
+
+        Returns:
+            pd.Dataframe: The response data as a Pandas Data Frame plus viadot metadata.
+        """
+        data = kwargs.get("data", False)
+        column_suffix = kwargs.get("column_suffix", None)
+        drop_columns = kwargs.get("drop_columns", None)
```

Comment:
It seems like this is exactly the opposite of what the name says - it's the columns you want to keep? So it should be more like just `columns`.

Reply:
changed.